Fix offline problems

galfert opened the pull request:
Refs #69 

This fixes a few problems when being offline.

1. Grey/fade out the Dropbox and Google Drive icons the same way as the remoteStorage icon
2. Set the text of the widget to "Offline" instead of "Not connected"
3. Fix a problem that the widget wouldn't open when clicking the icon while being offline

The actual problem with Dropbox reported in #69 is fixed with remotestorage/remotestorage.js#1096.

skddc commented on src/widget.js:
```diff
@@ -135,15 +135,16 @@ Widget.prototype = {
       if (this.closed && state !== 'close') {
         this.rsWidget.classList.add('rs-state-close');
+      } else if (!this.closed) {
+        this.rsWidget.classList.remove('rs-state-close');
```
I don't see how this makes sense. The current class is already being removed in line 134, so why do we have to remove it again? This being unnecessary would also explain why the `else if` just looks strange when reading it without context.

galfert replied:
Yes, the `rs-state-close` exists in addition to the other state (e.g. `rs-state-connected` or `rs-state-offline`. I did not code the state system, so I don't know how it was intended. But how else would you indicate that the widget is offline AND closed?

skddc replied:
Ok, I see. But that still doesn't explain the code. Also, it's something we should probably change right now, so that it's both cleaner and easier to understand and maintain.

galfert replied:
Yes, I think it would be better to not handle the closing of the widget using a state class, but a different class (e.g. `rs-closed`) instead. That way we can remove the logic from the `state()` function and just add or remove the class from the `open()` and `close()` functions.

galfert commented:
I changed the widget to use a separate class `rs-closed` for indicating a closed widget.
Similarly, I also added two methods `setOffline() and `setOnline()` and a class `rs-offline`, because the offline state was also handled differently from the other states in `setState()` and was also mixed with the other states. 

So now there should never be two `rs-state-x` classes active at the same time.

skddc commented:
Great!

skddc commented:
I tested this, and it all works as advertised.

(Unfortunately, I found that the Dropbox error is still there, even with the 1.0.1 rs.js release. See #69.)
